src/common/Requirements.ts

```diff
@@ -86,54 +88,64 @@ export abstract class BaseSetup implements RequirementList {
-                tests: []
-            };
-            let count = 0;
-            results.forEach((result) => {
-                if (result.status === 'fulfilled') {
-                    testResult.tests.push({
-                        hasPassed: true,
-                        message: result.v,
-                        testResult: 'Passed'
-                    });
-                } else if (result.status === 'rejected') {
-                    testResult.hasMetAllRequirements = false;
-                    testResult.tests.push({
-                        hasPassed: false,
-                        message: result.e,
-                        testResult: 'Failed'
-                    });
-                }
-                count++;
-            });
-
-            const tree = cli.tree();
-            tree.insert('Setup');
-            testResult.tests.forEach((test) => {
-                tree.nodes.Setup.insert(
-                    `${
-                        test.hasPassed
-                            ? chalk.bold.green(test.testResult)
-                            : chalk.bold.red(test.testResult)
-                    }: ${
-                        test.hasPassed
-                            ? chalk.green(test.message)
-                            : chalk.red(test.message)
-                    }`
-                );
-            });
-            tree.display();
-            return Promise.resolve(testResult);
+        const testResult: SetupTestResult = {
+            hasMetAllRequirements: true,
+            tests: []
+        };
+
+        let totalDuration: number = 0;
+        let stepDuration: number = 0;
+        const obs = new PerformanceObserver((items) => {
+            stepDuration = items.getEntries()[0].duration / 1000;
+            performance.clearMarks();
         });
+        obs.observe({ entryTypes: ['measure'] });
+
+        for (const requirement of this.requirements) {
+            stepDuration = 0;
+            performance.mark('Start');
+            const wrappedPromise = WrappedPromise(requirement.checkFunction());
+            const result = await wrappedPromise;
+            performance.mark('End');
+            performance.measure('StepDuration', 'Start', 'End');
+            totalDuration += stepDuration;
+            if (result.status === 'fulfilled') {
+                testResult.tests.push({
+                    duration: stepDuration,
+                    hasPassed: true,
+                    message: result.v,
+                    testResult: 'Passed'
+                });
+            } else if (result.status === 'rejected') {
+                testResult.hasMetAllRequirements = false;
+                testResult.tests.push({
+                    duration: stepDuration,
+                    hasPassed: false,
+                    message: result.e,
+                    testResult: 'Failed'
+                });
+            }
+        }
+
+        const setupMessage = `Setup (${totalDuration.toFixed(3)} sec)`;
```

Comment:
Don't forget to do an `obs.disconnect()` somewhere down here, once you're done with it.